Gallery Activity + Improvement for download activity

stefanoromanello opened the pull request:
This is a gallery for the taken images (not for the testing images).
It uses the meta-tags of the image for storing all the important informations and fragments for managing the main gallery and the detail view page.

For this code I needed the last downloadDbActivity Commit so this pull request includes also the last version of the download Activity which is already in the common demo but this new version adds some few permission verifications

lucamoroz committed:
Removed hardcoded strings

PietroPrandini committed:
Initial commit

PietroPrandini committed:
Dir

PietroPrandini committed:
removeMe.md added to preserve the directories structures on github

PietroPrandini committed:
README deleted

PietroPrandini committed:
.gitignore updated with the version contained in the CommonDemo branch

PietroPrandini committed:
Gradle files

lucamoroz committed:
Updated javadoc and simplified code

PietroPrandini committed:
Preparing branch for the code review

lucamoroz committed:
Photo testing alteration implementation (#48)

* added test photo alteration compatibility - PhotoTester.testAndReport() will return a report which includes the evaluation of any alteration of a test. Plus improved specs and jUnit tests

* added get alteration tag method in TestElement

* First commit

* First commit

* details

* Forced portrait mode to TestAlterationsActivity

* better exception handling, improved specs

* fixed null pointer exception in method run()

* Merge with AdapterAlterationsTest

* removed try - catch block

* removed try - catch block

* OnClickListener

* Removed button in adapterTestAlterations

* Log

* fixed setting bitmap alteration

* Optimized PhotoTester constructor

* Removed hardcoded strings

* Updated javadoc and simplified code

lucamoroz committed:
Merge branch 'photo_testing' of https://github.com/PrandiniUniPD/OCRCamera into photo_testing

BalzPit committed:
Performance Improvement

Created the class BrightnessRecognition to run the recognition on a separate thread to improve performance. The class is used in "cameraActivity", after the thread is opened

lucamoroz committed:
Added comments - replaced Semaphore availableThreads with a FixedThreadPool

frankplus committed:
Renamed to comply style document

frankplus committed:
Some performance improvements

frankplus committed:
Some performance improvements

frankplus committed:
Some changes

BalzPit committed:
minor changes

PietroPrandini committed:
AdapterTestElement with alterations summary + TestResultActivity more readable for reviews

PietroPrandini committed:
TestResultActivity more readable for reviews

BalzPit committed:
ready for code review

PietroPrandini committed:
Changes requested from the review

PietroPrandini committed:
Changes requested

BalzPit committed:
Merge branch 'CommonDemo' into Brightness_recognition

lucamoroz committed:
Added edited photos tags stats

lucamoroz committed:
fixed return missing

lucamoroz committed:
fixed alterations tags gain

frankplus committed:
added comments and other changes made during code review

BalzPit committed:
Suggestions added to the code

added integer return value to imgBrightness and moved the toasts to be called inside the thread in takePhoto method of CameraActivity

lucamoroz committed:
Solved memory leak caused by loading all the photos on the heap memory. Now class TestElement stores the image path rather than the full image

PietroPrandini committed:
Removing the removeMe.md files

PietroPrandini committed:
Removing the removeMe.md files

frankplus committed:
Load inci database and created LevenshteinStringComparator

frankplus committed:
Show ingredients in ingredients activity using adapters

ingredientsActivity is started from resultActivit in the menu.

frankplus committed:
changed csv parsing

frankplus committed:
Added explicative comments

frankplus committed:
Added description and checking for empty line in csv parsing

frankplus committed:
Better visualization, showing and sorting by similarity,now using thread

Better visualization
showing  similarity
Sorting by similarity
Moved processing into a thread
Showing progress Dialog

lucamoroz committed:
Covered NaN case in function ingredientsTextComparison

lucamoroz committed:
Covered NaN case in function ingredientsTextComparison

frankplus committed:
ingredients activity launch floating action button in ResultActivity 

moved button from menu into a floating action button in result activity

frankplus committed:
Removing unnecessary code

frankplus committed:
Changed button color into white

lucamoroz committed:
Merge branch 'CommonDemo' into photo_testing

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/unipd/se18/ocrcamera/AdapterTestAlterations.java
#	app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java
#	app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java
#	app/src/main/java/unipd/se18/ocrcamera/TestElement.java

lucamoroz committed:
Solved conflicts with CommonDemo

frankplus committed:
Merge branch 'CommonDemo' into IngredientsDisplay

stefanoromanello committed:
Added Login for FTP Credentials

stefanoromanello committed:
Fixed login error messages

stefanoromanello committed:
Check folder exist

stefanoromanello committed:
Check internet connection

frankplus committed:
Added similarity threshold condition

frankplus committed:
Added junit test

frankplus committed:
ignoring case during text search in db

PrandiniUniPD committed:
Improving tests views, starting.

PrandiniUniPD committed:
Cleaning

PrandiniUniPD committed:
Alterations programmatically if needed

PrandiniUniPD committed:
Formatting alterations

PrandiniUniPD committed:
Cleaning

PrandiniUniPD committed:
Details of alterations view

PrandiniUniPD committed:
Cleaning

PrandiniUniPD committed:
Details

PrandiniUniPD committed:
Progress bar, working on

PrandiniUniPD committed:
details of code styling

frankplus committed:
Renamed Levenshtein Class and added interface

stefanoromanello committed:
Fixed permission request

BalzPit committed:
Removed constructor and added static method implementation

lucamoroz committed:
Fixed progress bar update

stefanoromanello committed:
Fixed Documents Directory

stefanoromanello committed:
Better code for manage login errors

frankplus committed:
Small corrections

frankplus committed:
WIP: Started Text Auto Correction class and added constructor

lucamoroz committed:
Separated method saveReportToFile from testAndReport

frankplus committed:
Merge branch 'photo_testing' into textAutoCorrection

leopardo-rossi-developer committed:
Class initial structure

Definition of the methods that will be used to retrieve and store images for a gallery activity

frankplus committed:
WIP: Implemented text correction

frankplus committed:
Solved dependency conflict and few changes to auto correction

frankplus committed:
Now showing in test result details

lucamoroz committed:
Added listener to show progress on TestResultActivity

leopardo-rossi-developer committed:
Get images method

Implementation of the method to get images from data directory with metadata

frankplus committed:
Added new search ingredient algorithm using text auto correction class

leopardo-rossi-developer committed:
Store image method

Implementation of the method to store an image into data directory with metadata

leopardo-rossi-developer committed:
Exception handling

Exception handled during the storing process

stefanoromanello committed:
Created the gallery Activity

Created the gallery activity and all the related Adpters. 
In this commit for testing purposes the app starts with the gallery

leopardo-rossi-developer committed:
Code restyling

Deleting some unnecessary part of code

leopardo-rossi-developer committed:
PhotoStructure update

Reliability field added

leopardo-rossi-developer committed:
Reliability metadata tag

Reliability metadata tag addition

frankplus committed:
small changes

stefanoromanello committed:
[TO EDIT] Draft part of the fragment manager.

frankplus committed:
Change of algorithm now using bkTrees for  approximate string search 

Also deleted StringDistance interface

frankplus committed:
Added some comments

leopardo-rossi-developer committed:
Comments

stefanoromanello committed:
Improved Fragment + manage navigation throughout fragments

stefanoromanello committed:
Merge branch 'downloadPhotosFromDb' into galleryActivity

stefanoromanello committed:
Fixed storage Permission

frankplus committed:
Merge branch 'photo_testing' into textAutoCorrection

stefanoromanello committed:
Implemented delete action

stefanoromanello committed:
Added Fragment tag to Strings

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/galleryActivity' into galleryActivity

stefanoromanello committed:
Fixed Back Button

stefanoromanello committed:
Fixed close fragment method

stefanoromanello committed:
Fixed back Fragment from button

onBackPressed wasn't detecting the correct FragmentManager

leopardo-rossi-developer committed:
Metadata saving

Metadata are now saved with the image after it is shoot

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/galleryActivity' into galleryActivity

stefanoromanello committed:
Remove comments for delete images

leopardo-rossi-developer committed:
Changes in layout

stefanoromanello committed:
Fixed Layout

stefanoromanello committed:
Improved Performance

Resized images inside cardview

stefanoromanello committed:
Save image with ingredients + fixing some parts

stefanoromanello committed:
Use of formatted strings 

Avoid "?" letters in the results

frankplus committed:
Improved TEXT_SPLIT search method using bk-tree, and other changes

also: 
- removed similarity and found text from listview (printing them in debugging)
- Code refactoring and added some comments

frankplus committed:
small changes

frankplus committed:
separated different search method into two classes. Added some tests.

frankplus committed:
Added some comments

frankplus committed:
Better test report visualization

stefanoromanello committed:
Code improvement

Comments + few code parts

frankplus committed:
Now showing ingredients in result activity (IngredientsActivity removed)

Also some small changes

frankplus committed:
Added some junit tests, changed TextAutoCorrection structure

also now checking for the last word while correcting the text.

frankplus committed:
match longer names first and remove from text, added some JUnit tests

frankplus committed:
small changes

PrandiniUniPD committed:
Merge remote-tracking branch 'PietroPrandiniFork/codereview' into restructuring

# Conflicts:
#	README.md
#	app/build.gradle
#	app/src/main/java/unipd/se18/ocrcamera/TextExtractor.java

PrandiniUniPD committed:
Merge remote-tracking branch 'PietroPrandiniFork/codereviewB' into restructuring

PrandiniUniPD committed:
TextExtractor - updates from code reviews

PrandiniUniPD committed:
recognizer package

frankplus committed:
Manage line breaks 

- If there is a hyphen preceding a line break, then merge current line and next line togheter 
(e.g. "COPOL-
YMER" becomes "COPOLYMER").
- If there are line breaks (without hyphen) then substitute line break with a whitespace

frankplus committed:
Added more tests

frankplus committed:
correction of whitespaces before and after hyphen

frankplus committed:
Moved progress dialog set message into runOnUiThread

frankplus committed:
Display ingredients functions information

PrandiniUniPD committed:
New structure of OCR recognizing system adapted

PrandiniUniPD committed:
Processing string in ResultActivity

frankplus committed:
Showing average correct ingredients extracted in testing.

PrandiniUniPD committed:
Comments

PrandiniUniPD committed:
Details

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/photo_testing' into restructuring

# Conflicts:
#	app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java
#	app/src/main/java/unipd/se18/ocrcamera/TestListener.java
#	app/src/main/res/values/strings.xml

PrandiniUniPD committed:
Merging with photo_testing branch

frankplus committed:
TextSplitIngredientsExtractor now case insensitive search

PrandiniUniPD committed:
Details

frankplus committed:
Fixed image rotation problem reading exif metadata

frankplus committed:
Merge branch 'photo_testing' into textAutoCorrection

PrandiniUniPD committed:
Renaming

PrandiniUniPD committed:
Details

PrandiniUniPD committed:
Details

PrandiniUniPD committed:
Details

PrandiniUniPD committed:
Pic of alterations in test details

frankplus committed:
Added modularity and multithreading in ResultActivity. Added ProgressBar

stefanoromanello committed:
Improved the way to know if I've deleted an image

stefanoromanello committed:
Removed developing comments

stefanoromanello committed:
Fixed saveToFile name extension

frankplus committed:
Now using circular progress bar

TODO: show progress with circular progress bar

PrandiniUniPD committed:
Data by intent from TestsListAdapter to TestDetailsActivity

frankplus committed:
Circular progress bar showing progress

PrandiniUniPD committed:
AddViewBelow methods

PrandiniUniPD committed:
Alterations details programmatically if needed in the TestDetailsActivity

PrandiniUniPD committed:
Bold to the title added programmatically

frankplus committed:
Fixed a bug

frankplus committed:
Added colon as splitting symbol in TextSplitIngredientsExtractor

frankplus committed:
textcorrection: ignoring spaces before and after slash

frankplus committed:
ingredients search ignore non alphanumeric characters

BalzPit committed:
Moved to ResultActivity

The brightnessrecognition is now called inside ResultActivity and the thread is no longer anonimous. Added strings  to the resources to support localization of the toasts' text

BalzPit committed:
Merge branch 'CommonDemo' into Brightness_recognition

PrandiniUniPD committed:
Merge branch 'textAutoCorrection' into restructuring

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java
#	app/src/main/java/unipd/se18/ocrcamera/TestElementDetails.java
#	app/src/main/res/values/strings.xml

PrandiniUniPD committed:
Merging with the work of Francesco Pham

Nintenking commented:
Both the automatic login (with the .txt file in Documents) and the manual login work just fine; the gallery shows exactly all and only the last taken images via our app.  
This pull request is fine by me!

lucamoroz committed:
Refactored PhotoTester

PrandiniUniPD committed:
Merge branch 'galleryActivity' into restructuring

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java
#	app/src/main/res/values/strings.xml

PrandiniUniPD committed:
Merging with the work of Stefano Romanello and Leonardo Rossi

frankplus committed:
Created a Singleton for ingredient extractor 

Avoid using singleton expensive inci db load and extractor initialization every time a photo is taken

PrandiniUniPD committed:
Code adjusting about the testing section

PrandiniUniPD committed:
Test elements sorted in the TestListActivity

PrandiniUniPD committed:
Cleaning testing area

PrandiniUniPD committed:
Correctness colors adjusted

PrandiniUniPD committed:
details about TestUtils

frankplus committed:
Merge branch 'restructuring' into textAutoCorrection

frankplus committed:
Merge branch 'textAutoCorrection' into restructuring

frankplus committed:
Fixed list view being cut out

frankplus committed:
Merge remote-tracking branch 'origin/restructuring' into restructuring

frankplus committed:
Fixed progress bar not disappearing when no ingredients are found

frankplus committed:
small changes

frankplus committed:
small changes

lucamoroz committed:
Divided RunnableTest.run() into simpler methods (class PhotoTester)

frankplus committed:
moved ingredients extraction test in separate method, more modularity

frankplus committed:
Moved inci and ingredients extraction stuff into a different package

frankplus committed:
Merge remote-tracking branch 'origin/textAutoCorrection' into textAutoCorrection

frankplus committed:
small change

frankplus committed:
Merge branch 'textAutoCorrection' into restructuring

frankplus committed:
Reconstruct original order of ingredients and other small changes

PrandiniUniPD committed:
Comments

stefanoromanello committed:
Code improvements

frankplus committed:
Added some comments

frankplus committed:
comments

PrandiniUniPD committed:
TextRecognizer module

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/restructuring' into restructuring

frankplus committed:
Change singleton structure

frankplus committed:
Change singleton structure

lucamoroz committed:
Refactored ResultActivity, removed CountDownLatch and added an AsyncTask to update the UI

lucamoroz committed:
Added comments

frankplus committed:
Merge branch 'textAutoCorrection' into restructuring

frankplus committed:
Merge remote-tracking branch 'origin/textAutoCorrection' into textAutoCorrection

frankplus committed:
Parallelised inci loading with ocr extraction

frankplus committed:
Merge branch 'textAutoCorrection' into restructuring

frankplus committed:
Moved inci db loading into launcher activity, renamed navigator activity

frankplus committed:
Added comments

frankplus committed:
Updated camerakit version and Passing context to asynctask

frankplus committed:
Added wikipedia request in ingredient details activity

lucamoroz committed:
added menu search icon

lucamoroz committed:
created searchable item and search activity

lucamoroz committed:
Showing results as listview

lucamoroz committed:
Added on item click web search

lucamoroz committed:
Added comments

lucamoroz committed:
Added search with suggestions

lucamoroz committed:
Added async ingredients list load, added comments

frankplus committed:
Print analyzed text highlighting extracted ingredients

frankplus committed:
Merge branch 'textAutoCorrection' into wikipedia

frankplus committed:
substituted compile with implementation

frankplus committed:
small change in tests

frankplus committed:
change name NameMatchIngredientsExtractor to CamelCase

frankplus committed:
Merge branch 'textAutoCorrection' into wikipedia

Commit:
Fixed web search intent, fixed AutoCompleteTextView covering the button, handled the case intent action is not action search

frankplus committed:
Change text correction code, now including hyphen in word

frankplus committed:
Change text correction code, now including hyphen in word

frankplus committed:
Merge branch 'textAutoCorrection' into wikipedia

frankplus committed:
Merge branch 'search_ingredient_menu' into demo_g3

frankplus committed:
single time list ingredients load using singleton

frankplus committed:
Fix merge conflict

frankplus committed:
Renamed singleton class

frankplus committed:
PhotoTester using singleton for inci loading

frankplus committed:
Merge branch 'galleryActivity' into demo_g3

frankplus committed:
Added search button in ingred details activity.

frankplus committed:
change text correction into a faster and shorter code

frankplus committed:
add a test

frankplus committed:
Merge Brightness_recognition into demo_g3

moved brightness recognition thread into a separate class

frankplus committed:
changed BrightnessRecognitionThread into private

frankplus committed:
Changed BrightnessRecognition thread into an asynctask

frankplus committed:
Changed ingredient details activity into a dialog fragment

Commit:
Implemented search button, added on enter key listener to perform a search, other view improvements

Commit:
Fixed null pointer exception caused by selectedIngredient.getInciName

PrandiniUniPD committed:
Testing pics section - some bug fixes

frankplus committed:
Merge branch 'restructuring' into demo_g2+g3

frankplus committed:
Avoid hardcoding

frankplus committed:
added comments

frankplus committed:
Added close button to ingredient details fragment

frankplus committed:
using vector asset search icon

frankplus committed:
Merge branch 'search_ingredient_menu' into demo_g3

frankplus committed:
Merge branch 'wikipedia' into demo_g3

frankplus committed:
Merge branch 'textAutoCorrection' into demo_g3

stefanoromanello committed:
Merge branch 'demo_g3' into galleryActivity

stefanoromanello committed:
Added some Gallery's tests

stefanoromanello committed:
Used allergens found instead of reliability

stefanoromanello committed:
Removed reliability and allergens, implemented AdapterIngredient

stefanoromanello committed:
Comments

frankplus committed:
static asynctask and remove unnecessary code

PrandiniUniPD commented:
Outdated
